ingest/ingest_pipeline.py

```diff
@@ -487,17 +502,27 @@ def extract_from_anndata(self):
         )
         if self.anndata.validate():
             self.report_validation("success")
+            # process matrix data
+            ### TODO: how to associate "raw_count" cells to anndata file
+            if self.kwargs.get("extract") and "processed_expression" in self.kwargs.get(
+                "extract"
+            ):
+                self.anndata.process_matrix()
+            # Get cluster extraction parameters and perform extraction
             if self.kwargs.get("extract") and "cluster" in self.kwargs.get("extract"):
                 if not self.kwargs["obsm_keys"]:
-                    self.kwargs["obsm_keys"] = ['X_tsne']
+                    self.kwargs["obsm_keys"] = ["X_tsne"]
+                # TODO: perform check for successful extraction or report failure and exit
```

Comment:
Please note Jira ticket for TODO, e.g. `TODO (SCP-9876): perform...`".

Reply:
Done `492d032`